onnxruntime/contrib_ops/webgpu/quantization/dp4a_matmul_nbits.cc

```diff
@@ -249,11 +262,106 @@ Status DP4AMatMulNBitsProgram::GenerateShaderCode(ShaderHelper& shader) const {
   return Status::OK();
 }
 
+// scale_A components = 1, b components = 4, output components = 1
+Status DP4AMatMulNBitsSmallMProgram::GenerateShaderCode(ShaderHelper& shader) const {
+  shader.AddInput("input_a", ShaderUsage::UseUniform);
+  shader.AddInput("scales_a", ShaderUsage::UseUniform);
+  shader.AddInput("input_b", ShaderUsage::UseUniform);
+  shader.AddInput("scales_b", ShaderUsage::UseUniform);
+  shader.AddOutput("output", ShaderUsage::UseUniform | ShaderUsage::UseElementTypeAlias);
+
+  constexpr uint32_t tile_size_k_vec = 16;  // tile K in input_b.
+  // tile_size is the number of rows of b each workgroup process.
+  // tile_size_k_vec is the number of columns of b each workgroup process and each element is a vec4<u32>.
```

Comment:
tile_size_k_vec is the number of columns of b each workgroup thread process and each element is a vec4<u32>.